pkg/errorchecker/errorchecker.go

```diff
@@ -0,0 +1,53 @@
+package errorchecker
+
+import (
+	"fmt"
+	"regexp"
+
+	"gopkg.in/check.v1"
+)
+
+type errorMatchesChecker struct {
+	*check.CheckerInfo
+}
+
+var ErrorMessageMatcher check.Checker = errorMatchesChecker{
+	&check.CheckerInfo{Name: "ErrorMatches", Params: []string{"value", "regex"}},
+}
+
+func (checker errorMatchesChecker) Check(
+	params []interface{},
+	names []string,
+) (result bool, errStr string) {
+	if params[0] == nil {
```

Comment:
Would this panic if params is nil? Should we add an additional check to check its not nil?

Reply:
`params` can't be `nil` it is passed from `check` framework.
https://github.com/kastenhq/check/blob/v1/helpers.go#L197

So, I suppose, we can rely on that.